src/System/IO/Extra.hs

```diff
@@ -53,35 +58,32 @@ readFileBinary file = do
 
 -- Strict file reading
 
+-- | A strict version of 'hGetContents'.
```

Comment:
In future you may want to split things that refactor existing functionality and add new functionality into separate commits - that way if I decide to accept only some of the pieces I can cherry pick them across without conflicting on the bits I copy by hand.